packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -432,6 +442,65 @@ class _ReorderableListViewState extends State<ReorderableListView> {
           }
           break;
       }
+    } else if (widget.header == null && widget.footer != null) {
+      headerPadding = EdgeInsets.zero;
+      switch (widget.scrollDirection) {
+        case Axis.horizontal:
+          if (widget.reverse) {
+            listPadding = EdgeInsets.fromLTRB(0, padding.top, padding.right, padding.bottom);
+            // Footer on the left
+            footerPadding = EdgeInsets.fromLTRB(padding.left, padding.top, 0, padding.bottom);
+          } else {
+            listPadding = EdgeInsets.fromLTRB(padding.left, padding.top, 0, padding.bottom);
+            // Footer on the right
+            footerPadding = EdgeInsets.fromLTRB(0, padding.top, padding.right, padding.bottom);
+          }
+          break;
+        case Axis.vertical:
+          if (widget.reverse) {
+            listPadding = EdgeInsets.fromLTRB(padding.left, 0, padding.right, padding.bottom);
+            // Footer on the top
+            footerPadding = EdgeInsets.fromLTRB(padding.left, padding.top, padding.right, 0);
+          } else {
+            listPadding = EdgeInsets.fromLTRB(padding.left, padding.top, padding.right, 0);
+            // Footer on the bottom
+            footerPadding = EdgeInsets.fromLTRB(padding.left, 0, padding.right, 0);
+          }
+          break;
+      }
+    } else {
+      switch (widget.scrollDirection) {
+        case Axis.horizontal:
+          if (widget.reverse) {
+            // Header on the right
+            headerPadding = EdgeInsets.fromLTRB(0, padding.top, padding.right, padding.bottom);
+            listPadding = EdgeInsets.fromLTRB(padding.left, padding.top, 0, padding.bottom);
+            // Footer on the left
+            footerPadding = EdgeInsets.fromLTRB(0, padding.top, 0, padding.bottom);
+          } else {
+            // Header on the left
+            headerPadding = EdgeInsets.fromLTRB(padding.left, padding.top, 0, padding.bottom);
+            listPadding = EdgeInsets.fromLTRB(0, padding.top, 0, padding.bottom);
+            // Footer on the right
+            footerPadding = EdgeInsets.fromLTRB(0, padding.top, padding.right, padding.bottom);
+          }
+          break;
+        case Axis.vertical:
+          if (widget.reverse) {
+            // Header on the bottom
+            headerPadding = EdgeInsets.fromLTRB(padding.left, 0, padding.right, padding.bottom);
+            listPadding = EdgeInsets.fromLTRB(padding.left, padding.top, padding.right, 0);
+            // Footer on the top
+            footerPadding = EdgeInsets.fromLTRB(padding.left, 0, padding.right, 0);
+          } else {
+            // Header on the top
+            headerPadding = EdgeInsets.fromLTRB(padding.left, padding.top, padding.right, 0);
+            listPadding = EdgeInsets.fromLTRB(padding.left, 0, padding.right, 0);
+            // Footer on the bottom
+            footerPadding = EdgeInsets.fromLTRB(padding.left, 0, padding.right, 0);
+          }
+          break;
+      }
```

Comment:
The interaction of this new footer with the existing header, padding and reverse settings explodes the code here quite a bit. 

I wonder if a simpler way to express this would be to start with the listPadding set to the given padding. Then check to see if the header is present. If it is remove the appropriate side from the listPadding and set the headerPadding accordingly. Then do the same thing for the footer. That way you wouldn't need cases for header only/footer only/both header and footer. They could be handled independently. Not sure that is much simpler, perhaps this is just inherently complicated.